test/configure.test.ts

```diff
@@ -0,0 +1,462 @@
+ *--------------------------------------------------------------------------------------------*/
+
+// The module 'assert' provides assertion methods from node
+import * as assert from 'assert';
+import * as assertEx from './assertEx';
+import * as vscode from 'vscode';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { Platform } from "../configureWorkspace/config-utils";
+import { ext } from '../extensionVariables';
+import { Suite } from 'mocha';
+import { configure } from '../configureWorkspace/configure';
+import { TestUserInput } from 'vscode-azureextensionui';
+import { globAsync } from '../helpers/async';
+import { getTestRootFolder, constants } from './global.test';
+
+let testRootFolder: string = getTestRootFolder();
+
+suite("configure (Add Docker files to Workspace)", function (this: Suite): void {
+    this.timeout(10 * 1000);
+
+    const outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel('Docker extension tests');
+    ext.outputChannel = outputChannel;
+
+    async function testConfigureDocker(platform: Platform, ...inputs: (string | undefined)[]): Promise<void> {
+        // Set up simulated user input
+        inputs.unshift(platform);
+        const ui: TestUserInput = new TestUserInput(inputs);
+        ext.ui = ui;
+
+        await configure(testRootFolder);
+        assert.equal(inputs.length, 0, 'Not all inputs were used.');
+    }
+
+    async function writeFile(subfolderName: string, fileName: string, text: string): Promise<void> {
+        await fse.ensureDir(path.join(testRootFolder, subfolderName));
+        await fse.writeFile(path.join(testRootFolder, subfolderName, fileName), text);
+    }
+
+    function fileContains(fileName: string, text: string): boolean {
+        let contents = fse.readFileSync(path.join(testRootFolder, fileName)).toString();
+        return contents.indexOf(text) >= 0;
+    }
+
+    function assertFileContains(fileName: string, text: string): void {
+        assert(fileContains(fileName, text), `Expected to find '${text}' in file ${fileName}`);
+    }
+
+    function assertNotFileContains(fileName: string, text: string): void {
+        assert(!fileContains(fileName, text), `Unexpected found '${text}' in file ${fileName}`);
+    }
+
+    async function getProjectFiles(): Promise<string[]> {
+        let files = await globAsync('**/*', {
+            cwd: testRootFolder,
+            dot: true, // include files beginning with dot
+            nodir: true
+        });
+        return files;
+    }
+
+    function testInEmptyFolder(name: string, func: () => Promise<void>): void {
+        test(name, async () => {
+            // Delete everything in the root testing folder
+            assert(path.basename(testRootFolder) === constants.testOutputName, "Trying to delete wrong folder");;
+            await fse.emptyDir(testRootFolder);
+            await func();
+        });
+    }
+
+    // Node.js
+
+    suite("Node.js", () => {
```

Comment:
![screen shot 2018-07-17 at 1 32 09 pm](https://user-images.githubusercontent.com/11282622/42843733-d46f3d72-89c5-11e8-986b-f4af6b457bb6.png)

Not a good sign when adding a new file - can we split it up by language maybe?